### DISCUSSION TITLE
Error: cron is not running

### DISCUSSION THREAD

**Discussion by greenbamboo** [friendly-bits/geoip-shell#20] [Q&A] [answered]
Hello!

Thanks for this nice program. I recently installed it on some systems and it works great, however at two servers the installation fails with the message:
Error: cron is not running.

Both systems are Ubuntu servers. (Ubuntu 20.04 and Ubuntu 22.04) Cron is running on both.
The Geoip-shell version is: 0.5.9

Any idea what I can do?

Thank you

**Comment by friendly-bits** [comment]
Hi, please run the install script with the '-d' option to enable debug messages and post the output here.

**Comment by friendly-bits** [comment]
Also please post the output of these commands:

```
sudo systemctl status cron.service
echo "crontab:"; sudo crontab -u root -l; echo $?
echo "pidof:"; sudo pidof cron; echo $?
echo "command -v:"; sudo command -v cron; echo $?
echo "pgrep1:"; sudo pgrep -af "/cron"
echo "pgrep2:"; sudo pgrep -af "/cron" | awk -F' ' "BEGIN{rv=1} \$2 ~ /\/cron([ \t]|\$)/ {print \$2; rv=0; exit} END{exit rv}"; echo $?
```

**Comment by greenbamboo** [comment]
Thank you for your quick response.
Here the output of the commands:

`sh geoip-shell-install.sh -d`
Started *INSTALL* with args: '-d' 
Debug: install: Detected init: 'systemd'.
install: Cleaning up previous installation (if any)...
Started *UNINSTALL* with args: '-r' 
Started *UNINSTALL* with args: '-r' 
Deleting geoip-shell main scripts from /usr/bin...
Deleting the library scripts directory '/usr/lib/geoip-shell'...
Uninstall complete.

Back to *UNINSTALL*...
Back to *INSTALL*...
Copying scripts to /usr/bin... Ok.
Copying library scripts to /usr/lib/geoip-shell... Ok.
Started *MANAGE* with args: 'configure' 
Debug: manage: check_cron: $no_persist is 'false'. $cron_rv is ''.
Debug: manage: check_cron: Reading crontab failed.
manage: Error: cron is not running.
manage: The cron service needs to be enabled and started in order for persistence and automatic ip list updates to work.
manage: If you want to use geoip-shell without persistence and automatic ip list updates,
manage: install/configure geoip-shell with options '-n' '-s disable'.

Would you like geoip-shell to enable and start the cron service on this device? [y|n].
y|n:


`sudo systemctl status cron.service`
● cron.service - Regular background program processing daemon
     Loaded: loaded (/lib/systemd/system/cron.service; enabled; vendor preset: enabled)
     Active: active (running) since Sat 2024-10-19 00:24:56 CEST; 2 days ago
       Docs: man:cron(8)
   Main PID: 972 (cron)
      Tasks: 1 (limit: 4563)
     Memory: 1.4M
        CPU: 792ms
     CGroup: /system.slice/cron.service
             └─972 /usr/sbin/cron -f -P

Okt 21 18:17:01 server4 CRON[6375]: (root) CMD (   cd / && run-parts --report /etc/cron.hourly)
Okt 21 18:17:01 server4 CRON[6374]: pam_unix(cron:session): session closed for user root
Okt 21 19:17:01 server4 CRON[6389]: pam_unix(cron:session): session opened for user root(uid=0) by (uid=0)
Okt 21 19:17:01 server4 CRON[6389]: pam_unix(cron:session): session closed for user root
Okt 21 20:17:01 server4 CRON[6401]: pam_unix(cron:session): session opened for user root(uid=0) by (uid=0)
Okt 21 20:17:01 server4 CRON[6401]: pam_unix(cron:session): session closed for user root
Okt 21 20:37:01 server4 CRON[6411]: pam_unix(cron:session): session opened for user root(uid=0) by (uid=0)
Okt 21 20:37:01 server4 CRON[6411]: pam_unix(cron:session): session closed for user root
Okt 21 21:17:01 server4 CRON[6438]: pam_unix(cron:session): session opened for user root(uid=0) by (uid=0)
Okt 21 21:17:01 server4 CRON[6438]: pam_unix(cron:session): session closed for user root

`echo "crontab:"; sudo crontab -u root -l; echo $?`
crontab:
no crontab for root
1

`echo "pidof:"; sudo pidof cron; echo $?`
pidof:
972
0

`echo "command -v:"; sudo command -v cron; echo $?`
(sudo command didn't work so I typed as root: **echo "command -v:"; command -v cron; echo $?**

command -v:
/usr/sbin/cron
0

`echo "pgrep1:"; sudo pgrep -af "/cron"`
pgrep1:
972 /usr/sbin/cron -f -P
6913 sudo pgrep -af /cron
6914 sudo pgrep -af /cron

`echo "pgrep2:"; sudo pgrep -af "/cron" | awk -F' ' "BEGIN{rv=1} \$2 ~ /\/cron([ \t]|\$)/ {print \$2; rv=0; exit} END{exit rv}"; echo $?`
pgrep2:
/usr/sbin/cron
0

**Reply by friendly-bits** [reply, marked as answer]
> no crontab for root

This means that the initial crontab file has not been created which is why you are getting this error message. Currently geoip-shell does not implement specific error message for this case, so it just says 'cron is not running'. Perhaps I should fix this. Anyway, geoip-shell should be able to solve this automatically if you enter 'y' in the dialog. Did you try that?

**Comment by greenbamboo** [comment]
Ah, I see. I didn't try to press y but I just created a test cron job and now the installer works. Thank you!

**Reply by greenbamboo** [reply]
On the second machine I pressed y and now it also works. Thanks again

**Reply by friendly-bits** [reply]
Great! Thank you for reporting this - I will improve the error message in the next version.

**Reply by friendly-bits** [reply]
v0.5.10 implements more granular error and info messages for this case
